### ISSUE TITLE
Redirect to another entryPoint gives "Too many redirects" error

### ISSUE THREAD

**Issue by carlosedp** [containous/traefik#2925]
### Do you want to request a *feature* or report a *bug*?

Bug

### What did you do?

I want to redirect a specific Ingress from http to https in a Kubernetes Cluster. Currently using v1.5.2 image.

My Ingress yaml is:

```yml
apiVersion: extensions/v1beta1
kind: Ingress
metadata:
  name: kubernetes-dashboard-external
  namespace: kube-system
  annotations:
    traefik.frontend.rule.type: PathPrefix
    traefik.frontend.redirect.entryPoint: https
    ingress.kubernetes.io/auth-type: basic
    ingress.kubernetes.io/auth-secret: dashboard-auth
spec:
  rules:
  - host: dashboard.cloud.mydomain.com
    http:
      paths:
      - path: /
        backend:
          serviceName: kubernetes-dashboard
          servicePort: 80
```

When I access the address, I get a "too many redirects" error.

```
$ wget http://dashboard.cloud.mydomain.com
--2018-02-24 16:10:44--  http://dashboard.cloud.mydomain.com/
Resolving dashboard.cloud.mydomain.com (dashboard.cloud.mydomain.com)... 190.219.173.145
Connecting to dashboard.cloud.mydomain.com (dashboard.cloud.mydomain.com)|190.219.173.145|:80... connected.
HTTP request sent, awaiting response... 302 Found
Location: https://dashboard.cloud.mydomain.com:443/ [following]
--2018-02-24 16:10:44--  https://dashboard.cloud.mydomain.com/
Connecting to dashboard.cloud.mydomain.com (dashboard.cloud.mydomain.com)|190.219.173.145|:443... connected.
HTTP request sent, awaiting response... 302 Found
Location: https://dashboard.cloud.mydomain.com:443/ [following]
....
20 redirections exceeded.
```

### What did you expect to see?

Correct redirection from http to https

### What did you see instead?

Too many redirects error.

As a workaround I replaced the annotation:

`    traefik.frontend.redirect.entryPoint: https`

With:

```
    #traefik.frontend.redirect.regex: ^http://(.*)
    #traefik.frontend.redirect.replacement: https://$1
```

and have a functional deployment.

### Output of `traefik version`: (_What version of Traefik are you using?_)

```
Version:      v1.5.2
Codename:     cancoillotte
Go version:   go1.9.4
Built:        2018-02-12_10:56:31AM
OS/Arch:      linux/arm64
```

### What is your environment & configuration (arguments, toml, provider, platform, ...)?

```toml
defaultEntryPoints = ["http","https"]
debug = false
logLevel = "INFO"

[entryPoints]
  [entryPoints.http]
  address = ":80"
  compress = true
  [entryPoints.https]
  address = ":443"
    [entryPoints.https.tls]

[web]
  address = ":8080"

[metrics]
  [metrics.prometheus]
  buckets=[0.1,0.3,1.2,5.0]
  entryPoint = "traefik"

[acme]
email = "aaa.@aaa.com"
storage = "/etc/traefik/acme.json"
acmeLogging = true
entryPoint = "https"
onDemand = false
OnHostRule = true
caServer = "https://acme-v01.api.letsencrypt.org/directory"
[acme.httpChallenge]
entryPoint="http"
```

**Comment by dtomcej** [comment]
Because you have:
`defaultEntryPoints = ["http","https"]`
All your new ingresses default to listen on both of the entrypoints (http and https).

Because you specify:
`traefik.frontend.redirect.entryPoint: https` in your ingress, you are blindly redirecting all requests to  the https entrypoint.

Therefore your request paths are:

Http request -> traefik -> Https redirect
Https request -> traefik -> Https redirect

And that continues until the limit is reached.
If you are wanting to ensure that your requests are using https, you should use one of the other methods to force an ssl-request, such as:

`ingress.kubernetes.io/ssl-redirect:true` or`ingress.kubernetes.io/ssl-temporary-redirect:true`

**Comment by carlosedp** [comment]
Ok, the point is that I use both http and https entrypoints... so I can't redirect the traffic globally to https. The regex is working fine. Gonna wait for the fix.
Thanks!

**Comment by traefiker** [comment]
Closed by #2929.
